### ISSUE TITLE
Adding more acceptable sources in ImportFactor.from

### ISSUE THREAD

**Issue by Siemienik** [Siemienik/xlsx-import#52]
As @Metastasis mentioned in https://github.com/Siemienik/xlsx-import/issues/4#issuecomment-711051787. Current factory has possibility only for reading workbook from local file system, what can't works under browser because security purposes.

# Actual 

The `ImportFactory.from` accept only filePath which is a string:

https://github.com/Siemienik/xlsx-import/blob/90c2b76564bd87433de19a9e8f55b9e064d14e09/src/ImporterFactory.ts#L6-L11

# Expected

ImportFactory should recognise type of argument and knowing this choose how to handle it.

| Type | What is | Should do |
|------|---------|-----------|
| `string` | File path | If node: read file from localsystem |
| `string` | File url | Download from url and read file as ArrayBuffer |
| `string` | Data url | Get ArrayBuffer |
| `string` | Buffer representation * | Transform to ArrayBuffer |
| `Buffer\|ArrayBuffer` | Buffer | Read indirectly from buffer |
| none of above | unknown  | I haven't idea yet, probably should throw an error |

* As @Metastasis mention:
> I realized that string can also be a buffer so probably my implementation is broken (string can represent buffer as well as file path).

**Comment by Metastasis** [comment]
@Siemienik 
This issue looks fine but
I'm not sure about File Url part since It looks for me like too much responsibility. From another point of view It could be a lot of copy-paste code in the client therefore this feature should be implemented
I think we could implement this and see what happens next:) (I assume somebody would create an issue if something will be broken)

**Comment by Siemienik** [comment]
>I'm not sure about File Url part since It looks for me like too much responsibility
Right, I considered about that. May we should implement first version without this and add it if requested. WDYT?

**Comment by Metastasis** [comment]
@Siemienik 
Yeah, I think we could omit File Url

**Comment by Siemienik** [comment]
connected with #196
